### PR TITLE
2290 sqlite db config

### LINKED ISSUE #2290: Configuration option for database path
## Is your feature request related to a problem? Please describe 👀

The request was made to be able to specify the path of the sqlite database, if sqlite is being used.

In mSupply currently, a common configuration is to have the datafile on a separate drive, to allow separation from the program files, enable simpler backup configuration and to have the data stored on a drive separate to the OS, which in theory allows for faster access.

## Describe the solution you'd like 🎁
Add an optional configuration setting to the `yaml` files, so specify the path of the database. This will be ignored if postgreSQL is being used.

If set, the SQLite file is loaded from the path, if not the current setting is used: which is the path of the executing binary.

Perhaps you'd like to do #817 at the same time?

### PR DESCRIPTION
**Pull request by fergie-nz**
Fixes #2290

# 👩🏻‍💻 What does this PR do? 
Adds optional database path in yaml file. This allows specification of database path

# 🧪 How has/should this change been tested? 
Add a path or nested path to local.yaml file, then start the back end with cargo run.

You should see the sqlite database set up in the specified path.

### CODE REVIEW

File: server/repository/src/database_settings.rs [hunk 1]
```diff
@@ -72,8 +73,19 @@ impl DatabaseSettings {
...
+                format!("{}/{}", path, self.connection_string())
+            }
+            None => self.connection_string(),
+        };
+        return result;
+    }
+
     pub fn connection_string_without_db(&self) -> String {
-        self.connection_string()
+        self.database_path()
```
Comment (jmbrunskill) [inline server/repository/src/database_settings.rs:88]: Does this work with postgres?

Comment (jmbrunskill) [reply server/repository/src/database_settings.rs:88]: Sorry I see it's behind the `#[cfg(all(not(feature = "postgres"), not(feature = "memory")))]` so I think it should be ok.

Comment (jmbrunskill) [reply server/repository/src/database_settings.rs:88]: but you will need to add this to the postgres `get_test_db_settings`
```
error[E0063]: missing field `database_path` in initializer of `DatabaseSettings`
  --> repository/src/test_db/postgres.rs:11:5
   |
11 |     DatabaseSettings {
   |     ^^^^^^^^^^^^^^^^ missing `database_path`
```

Try this PR with `cargo test --features=postgres` and probably start up the app in postgres mode as well for good measure :)

**Review by jmbrunskill — CHANGES_REQUESTED** [review]
`cargo test --features=postgres` isn't working, would be good to have that fixed before we merge.

File: server/repository/src/database_settings.rs (hunk 1 above)
Comment (clemens-msupply) [reply server/repository/src/database_settings.rs:88]: This `connection_string_without_db` is only used for the postgres feature right? could you please try to remove it for sqlite and memory? think it should work without it...

**Review by jmbrunskill — APPROVED** [review]
Looks good thanks Ferg!

**Pull request merged**